Lib/pdb.py

```diff
@@ -493,6 +493,15 @@ def _cmdloop(self):
             except KeyboardInterrupt:
                 self.message('--KeyboardInterrupt--')
 
+    def _update_file_mtime(self):
+        """update the file mtime table with the current frame's file"""
+        try:
+            filename = self.curframe.f_code.co_filename
+            mtime = os.path.getmtime(filename)
+            self._file_mtime_table.setdefault(filename, mtime)
```

Comment:
If not, then let's avoid going to the disk when the dict already has a value.

Reply:
Because we separated the update and check. We do not generate warnings on every update. We want to avoid `file change -> update(updated the new info) -> check(fail to find the change)`.

But yes, we should avoid updating it if not necessary.